### PR TITLE
Skip tests for "drafts"

### PR DESCRIPTION
**Pull request by dacbd**
Feel free to reject.

Trying to reduce the action spam from pushing commits to PRs while they are a "draft" as it seems we all work on changes in drafts (which help communicate the progress of changes)

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -14,12 +14,20 @@ jobs:
...
-    - run: make test
+  test:
+    if: ${{ github.event_name != 'pull_request' || github.event.pull_request.draft == false }}
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/setup-go@v2
+      with:
+        go-version: ^1.16
+    - uses: actions/checkout@v2
+    - run: make build && make test
```
Comment (0x2b3bfa0) [inline .github/workflows/test.yml:33]:
```suggestion
    needs: build
    if: ${{ ! github.event.pull_request.draft }}
    runs-on: ubuntu-latest
    steps:
    - uses: actions/setup-go@v2
      with:
        go-version: ^1.16
    - uses: actions/checkout@v2
    - run: make test
```

❓

**Comment by 0x2b3bfa0** [comment]
I usually (ab)use drafts to run tests and make sure my changes are green before marking them as ready for review. Nevertheless, it's just a questionable habit of mine, and it might make sense to disable tests to reduce noise until changes are ready. 🤷🏼‍♂️ 

🔔 @iterative/cml, what do you think?

**Comment by dacbd** [comment]
> I usually (ab)use drafts to run tests and make sure my changes are green before marking them as ready for review. Nevertheless, it's just a questionable habit of mine, and it might make sense to disable tests to reduce noise until changes are ready. 🤷🏼‍♂️
> 
> 🔔 @iterative/cml, what do you think?

I think that's totally fine and think that having the PRs open as drafts helps to communicate WIP, which I find helpful. I wouldn't put too much though into this and it definitely shouldn't try and change the way anyone works. I just like to see green ✔️, my idea was always build to catch typos but test when its done.

**Comment by casperdcl** [comment]
Closing for same reason as https://github.com/iterative/cml/pull/832#issuecomment-987848685 & https://github.com/iterative/terraform-provider-iterative/pull/325#issuecomment-988047236 though happy to continue discussion.

**Pull request closed without merging**